iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientConfiguration.java

```diff
@@ -208,10 +195,21 @@ private void configSasAuth(IotHubConnectionString iotHubConnectionString) {
 
     ClientConfiguration(IotHubConnectionString iotHubConnectionString, IotHubClientProtocol protocol, ClientOptions clientOptions)
     {
-        String mqttGatewayHostName = iotHubConnectionString.getMqttGatewayHostName();
-        if (mqttGatewayHostName != null && !mqttGatewayHostName.isEmpty())
+        String gatewayHostName = iotHubConnectionString.getGatewayHostName();
```

Comment:
This block of code appears twice in this class. Can you factor this out to a private method?